pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -37,131 +40,185 @@ func newTestManager() (*basicManager, *podtest.FakeMirrorClient) {
+				Namespace: metav1.NamespaceDefault,
+				Annotations: map[string]string{
+					kubetypes.ConfigSourceAnnotationKey: "api",
+					kubetypes.ConfigMirrorAnnotationKey: "mirror",
+				},
 			},
-		},
-	}
-	staticPod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			UID:         "123456789",
-			Name:        "bar",
-			Namespace:   "default",
-			Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "file"},
-		},
-	}
-
-	expectedPods := []*v1.Pod{
-		{
+		}
+		staticPod = &v1.Pod{
 			ObjectMeta: metav1.ObjectMeta{
-				UID:         "999999999",
-				Name:        "taco",
-				Namespace:   "default",
-				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "api"},
+				UID:         types.UID("static-pod-uid"),
+				Name:        "mirror-static-pod-name",
+				Namespace:   metav1.NamespaceDefault,
+				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "file"},
 			},
+		}
+		normalPod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("normal-pod-uid"),
+				Name:      "normal-pod-name",
+				Namespace: metav1.NamespaceDefault,
+			},
+		}
+	)
+
+	testCase := []struct {
+		name         string
+		podList      []*v1.Pod
+		wantPod      *v1.Pod
+		expectPods   []*v1.Pod
+		expectGetPod *v1.Pod
+		expectUID    types.UID
+	}{
+		{
+			name:         "Get normal pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      normalPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: normalPod,
+			expectUID:    types.UID("static-pod-uid"),
+		},
+		{
+			name:         "Get static pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      staticPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: staticPod,
+			expectUID:    types.UID("static-pod-uid"),
 		},
-		staticPod,
-	}
-	updates := append(expectedPods, mirrorPod)
-	podManager, _ := newTestManager()
-	podManager.SetPods(updates)
-
-	// Tests that all regular pods are recorded correctly.
-	actualPods := podManager.GetPods()
-	if len(actualPods) != len(expectedPods) {
-		t.Errorf("expected %d pods, got %d pods; expected pods %#v, got pods %#v", len(expectedPods), len(actualPods),
-			expectedPods, actualPods)
 	}
-	for _, expected := range expectedPods {
-		found := false
-		for _, actual := range actualPods {
-			if actual.UID == expected.UID {
-				if !reflect.DeepEqual(&expected, &actual) {
-					t.Errorf("pod was recorded incorrectly. expect: %#v, got: %#v", expected, actual)
+	for _, test := range testCase {
+		t.Run(test.name, func(t *testing.T) {
+			podManager, _ := newTestManager()
+			podManager.SetPods(test.podList)
+			actualPods := podManager.GetPods()
+			sort.Slice(actualPods, func(i, j int) bool {
+				return actualPods[i].Name < actualPods[j].Name
+			})
+			if diff := cmp.Diff(actualPods, test.expectPods); diff != "" {
+				t.Errorf("Test case %s failed: actualPods and expectPods are not equal", test.name)
```

Comment:
Since the test name is already passed to `t.Run()`, I don't think it's necessary to include the test name in the error message.

```suggestion
				t.Errorf("actualPods and expectPods are not equal: %s", diff)
```

Before

```console
--- FAIL: TestGetSetPods (0.00s)
    --- FAIL: TestGetSetPods/Get_normal_pod (0.00s)
        /Users/tsubasanagasawa/Documents/work/kubernetes/pkg/kubelet/pod/pod_manager_test.go:113: Test case Get normal pod failed: actualPods and expectPods are not equal
FAIL
FAIL	k8s.io/kubernetes/pkg/kubelet/pod	0.784s
FAIL
```

After

```console
--- FAIL: TestGetSetPods (0.00s)
    --- FAIL: TestGetSetPods/Get_normal_pod (0.00s)
        /Users/tsubasanagasawa/Documents/work/kubernetes/pkg/kubelet/pod/pod_manager_test.go:113: actualPods and expectPods are not equal:   []*v1.Pod{
              	&{ObjectMeta: {Name: "mirror-static-pod-name", Namespace: "default", UID: "static-pod-uid", Annotations: {"kubernetes.io/config.source": "file"}, ...}},
              	&{
              		TypeMeta: {},
              		ObjectMeta: v1.ObjectMeta{
            - 			Name:         "normal-pod-name",
            + 			Name:         "not-normal-pod-name",
              			GenerateName: "",
              			Namespace:    "default",
              			... // 12 identical fields
              		},
              		Spec:   {},
              		Status: {},
              	},
              }
FAIL
FAIL	k8s.io/kubernetes/pkg/kubelet/pod	0.682s
FAIL
```

Reply:
fixed